src/include/catalog/catalog_content.h

```diff
@@ -79,6 +81,7 @@ class CatalogContent {
     common::table_id_t nextTableID;
     std::unique_ptr<function::BuiltInFunctions> builtInFunctions;
     std::unordered_map<std::string, std::unique_ptr<function::ScalarMacroFunction>> macros;
+    common::FileSystem* fs;
```

Comment:
Can we construct the necessary `fileInfo` in constructor, so we don't need to store `fs`?

Reply:
As we discussed, we don't want to always hold the fd for the catalog while the database is runnning.